hs_composite_resource/tests/test_composite_resource_user_zone.py

```diff
@@ -73,25 +74,794 @@ def test_create_composite_resource_with_file_upload(self):
 
         # Deprecated: there should not be any GenericLogicalFile object at this point
         # Issue 2456 Create composite with uploaded file now part of logical file
-        self.assertEqual(GenericLogicalFile.objects.count(), 1)
-
-        # set the logical file - which get sets as part of the post resource creation signal
-        resource_post_create_actions(resource=self.composite_resource, user=self.user,
-                                     metadata=self.composite_resource.metadata)
+        self.assertEqual(GenericLogicalFile.objects.count(), 0)
 
         # there should be one resource at this point
         self.assertEqual(BaseResource.objects.count(), 1)
         self.assertEqual(self.composite_resource.resource_type, "CompositeResource")
         self.assertEqual(self.composite_resource.files.all().count(), 1)
         res_file = self.composite_resource.files.first()
+        # create the generic aggregation (logical file)
+        GenericLogicalFile.set_file_type(self.composite_resource, self.user, res_file.id)
 
         # check that the resource file is associated with GenericLogicalFile
+        res_file = self.composite_resource.files.first()
         self.assertEqual(res_file.has_logical_file, True)
         self.assertEqual(res_file.logical_file_type_name, "GenericLogicalFile")
+
         # there should be 1 GenericLogicalFile object at this point
         self.assertEqual(GenericLogicalFile.objects.count(), 1)
         self.composite_resource.delete()
 
+    def test_generic_aggregation_xml_files_creation(self):
+        """Test that aggregation metadata and map xml files are created on aggregation
+        creation"""
+
+        if not super(CompositeResourceTest, self).is_federated_irods_available():
+            return
+
+        # test that when we create composite resource with an uploaded file, then the uploaded file
+        # is automatically set to genericlogicalfile type
+        self.assertEqual(BaseResource.objects.count(), 0)
```

Comment:
Is this the right approach here?  This will throw a failure for this test, but in the case of failure, it really has nothing to do with the functionality in this test.

Reply:
Yeah it is not needed. Will remove.